util/bench/benchmarkSignVerify.cpp

```diff
@@ -0,0 +1,299 @@
+  string privateKeyFullPath(
+      {string(KEYS_BASE_PARENT_PATH) + string(EDDSA_KEYS_BASE_PATH) + string("/1/") + PRIV_KEY_NAME});
+
+  string privKey, pubkey, sig;
+
+  readFile(privateKeyFullPath, privKey);
+  readFile(publicKeyFullPath, pubkey);
+
+  const auto signingKey = getByteArrayKeyClass<EdDSAPrivateKey, EdDSAPrivateKeyByteSize>(privKey, KeyFormat::PemFormat);
+  auto signer_ = unique_ptr<TestTxnSigner>(new TestTxnSigner(signingKey.getBytes()));
+
+  // Sign with EdDSASigner.
+  size_t expectedSignerSigLen = signer_->signatureLength();
+  sig.reserve(expectedSignerSigLen);
+  size_t lenRetData;
+
+  uint64_t signaturesPerformed = 0;
+  {
+    picobench::scope scope(s);
+
+    for (int msgIdx = 0; msgIdx < s.iterations(); msgIdx++) {
+      sig = signer_->sign(randomData[msgIdx % RANDOM_DATA_ARRAY_SIZE]);
+      lenRetData = sig.size();
+      ++signaturesPerformed;
+      ConcordAssertEQ(lenRetData, expectedSignerSigLen);
+    }
+  }
+  s.set_result(signaturesPerformed);
+}
+
+/**
+ * @brief A benchmark which measures the time it takes for EdDSA verifier to verify a signature.
+ *
+ * @param s
+ */
+void edDSAVerifierBenchmark(picobench::state& s) {
+  string publicKeyFullPath(
+      {string(KEYS_BASE_PARENT_PATH) + string(EDDSA_KEYS_BASE_PATH) + string("/1/") + PUB_KEY_NAME});
+  string privateKeyFullPath(
+      {string(KEYS_BASE_PARENT_PATH) + string(EDDSA_KEYS_BASE_PATH) + string("/1/") + PRIV_KEY_NAME});
+
+  string privKey, pubkey, sig;
+
+  readFile(privateKeyFullPath, privKey);
+  readFile(publicKeyFullPath, pubkey);
+
+  const auto signingKey = getByteArrayKeyClass<EdDSAPrivateKey, EdDSAPrivateKeyByteSize>(privKey, KeyFormat::PemFormat);
+  const auto verificationKey =
+      getByteArrayKeyClass<EdDSAPublicKey, EdDSAPublicKeyByteSize>(pubkey, KeyFormat::PemFormat);
+
+  auto signer_ = unique_ptr<TestTxnSigner>(new TestTxnSigner(signingKey.getBytes()));
+  auto verifier_ = unique_ptr<TestTxnVerifier>(new TestTxnVerifier(verificationKey.getBytes()));
+
+  // Sign with EdDSASigner.
+  size_t expectedSignerSigLen = signer_->signatureLength();
+  sig.reserve(expectedSignerSigLen);
+  size_t lenRetData;
+
+  const auto offset = (uint8_t)rand() % RANDOM_DATA_ARRAY_SIZE;
+  sig = signer_->sign(randomData[offset]);
+  lenRetData = sig.size();
+  ConcordAssertEQ(lenRetData, expectedSignerSigLen);
+
+  uint64_t signaturesVerified = 0;
+  {
+    picobench::scope scope(s);
+
+    for (int msgIdx = 0; msgIdx < s.iterations(); msgIdx++) {
+      ++signaturesVerified;
+
+      // validate with EdDSAVerifier.
+      ConcordAssert(verifier_->verify(randomData[offset], sig));
+    }
+  }
+  s.set_result(signaturesVerified);
+}
+
+/**
+ * @brief A benchmark which measures the time it takes for RSA signer to sign a message.
+ *
+ * @param s
+ */
+void rsaSignerBenchmark(picobench::state& s) {
+  string publicKeyFullPath({string(KEYS_BASE_PARENT_PATH) + string(RSA_KEYS_BASE_PATH) + string("/1/") + PUB_KEY_NAME});
```

Comment:
Hard coded file paths are no more required because the key pairs are generated on the fly (using C++ key pair generation functions) and stored in variables and consumed for processing.